src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -308,10 +306,74 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
         BufferedReader bufferedReader = new BufferedReader(new FileReader(findFile));
         readFile(bufferedReader); // works with files that begin with ACT
         bufferedReader.close();
-      } catch (Exception e) {
+      } catch (IOException e) {
         //error
       }
+  }
+
+  /**
+   * This function fires when a user submits the testdata form. It loads test data if the user
+   * clicked the confirm button.
+   */
+  public void postAdminStatusButton(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException, PersistentDataStoreException {
+    String changeAdminStatusButton = request.getParameter("change_admin_status");
+    String editUsername = request.getParameter("username");
+    User editUser = userStore.getUser(editUsername);
+
+    if (editUser == null) {
+      request.setAttribute("error", "Not a user.");
+    } else {
+      if (changeAdminStatusButton.equals("promote")) {
+        if (editUser.getAdminStatus()) {
+          request.setAttribute("error", "User is already an admin.");
+        } else {
+          userStore.setIsAdmin(editUser, true);
+          request.setAttribute("success", "Promoted the user to admin!");
+        }
+      } else {
+        if (editUser.getAdminStatus()) {
+          userStore.setIsAdmin(editUser, false);
+          request.setAttribute("success", "Demoted the admin to user :(");
+        } else {
+          request.setAttribute("error", "User is already not an admin.");
+        }
+      }
+    }
+  }
+
+  /*
+  * Displays new messages from user of a selected text file & updates stats to
+  * include the new messages
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    String changeAdminStatusButton = request.getParameter("change_admin_status");
+    String confirmButton = request.getParameter("confirm");
+    String meUsername = (String) request.getSession().getAttribute("user");
+    User meUser = userStore.getUser(meUsername);
```

Comment:
maybe we can call this variable currentUser? and other related variables to use current instead of me